api/app/Models/PoolCandidate.php

```diff
@@ -261,6 +261,12 @@ public function assessmentResults(): HasMany
         return $this->hasMany(AssessmentResult::class);
     }
 
+    /** @return BelongsTo<AssessmentStep, $this> */
+    public function assessmentStep(): BelongsTo
+    {
+        return $this->belongsTo(AssessmentStep::class);
+    }
+
```

Comment:
Since this refers to "current assessment step", I think it is very important to be named as such so `currentAssessmentStep` 😛

Reply:
This is the name we are already using so I opted to keep it rather than update it everywhere else on the platform.

Reply:
Also, we only store a single step so adding current might insinute you could havbe a previous or next step. 

If we needed to reference some previous or next steps it might help, but a candidate can only be on a single step at a time so current seems a little redundant.

Reply:
I get both sides of this.  On one hand, "current" is redundant since there can only be one.  On the other hand, when reading through the code it isn't obvious what this relationship represents.  Could you maybe add a comment to the model file and the schema to help with that?